### PR TITLE
Pick up some of the shared workflows

### PR DESCRIPTION
**Pull request by mattmoor**
This picks up a shared definition of `DO NOT SUBMIT` to avoid divergent shell code.

This also picks up our KinD diagnostic dumping logic from downstream, which we've been using to debug assorted KinD-based failures.

I have a few more workflows on my list that we can share this way, and I'll likely keep adding them here.

@cpanato Not sure what the best way of getting this to use the dependabot form is, so PTAL and LMK how best to adjust this.  We haven't cut a release yet, but we can once we've gotten a bunch more of our workflows shared this way.

#### Ticket Link

N/A

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by dlorenc — APPROVED** [review]

**Comment by mattmoor** [comment]
I think we should probably snap a `v1` before we merge this (and/or pin to a commit), but I wanted to find out from @cpanato the best way to align with what we do here.

File: .github/workflows/donotsubmit.yaml [hunk 1]
```diff
@@ -17,39 +17,4 @@ jobs:
...
-          xargs grep -n "DO NOT SUBMIT" |
-          reviewdog -efm="%f:%l:%m" \
-                -name="DO NOT SUBMIT" \
-                -reporter="github-pr-check" \
-                -filter-mode="added" \
-                -fail-on-error="true" \
-                -level="error"
-
-          echo '::endgroup::'
+        uses: chainguard-dev/actions/donotsubmit@main
```
Comment (cpanato) [inline .github/workflows/donotsubmit.yaml:20]: can we use a git hash instead of main?

**Comment by cpanato** [comment]
@mattmoor, we need to add the following in the dependabot.yml definition

```yaml
  - package-ecosystem: "github-actions"
    directory: "/"
    schedule:
      interval: "daily"
```

then it will check the actions/jobs and update them according

File: .github/workflows/donotsubmit.yaml (hunk 1 above)
Comment (mattmoor) [reply .github/workflows/donotsubmit.yaml]: Done

**Comment by mattmoor** [comment]
Added the dependabot config as well.

File: .github/workflows/whitespace.yaml [hunk 2]
```diff
@@ -13,84 +13,11 @@ jobs:
...
-              echo $x:$((1 + $(wc -l $x | tr -s ' ' | cut -d' ' -f 1))): Missing newline
-            fi
-          done |
-          reviewdog -efm="%f:%l: %m" \
-                -name="EOF Newline" \
-                -reporter="github-pr-check" \
-                -filter-mode="added" \
-                -fail-on-error="true" \
-                -level="error"
-          echo '::endgroup::'
```
Comment (cpanato) [inline .github/workflows/whitespace.yaml:96]: nit/non blocking: new line

**Review by cpanato — APPROVED** [review]
cool!

File: .github/workflows/whitespace.yaml (hunk 2 above)
Comment (mattmoor) [reply .github/workflows/whitespace.yaml:96]: This was before 🤣  👍

File: .github/dependabot.yml [hunk 3]
```diff
@@ -20,3 +20,7 @@ updates:
   schedule:
     interval: daily
   open-pull-requests-limit: 10
+- package-ecosystem: "github-actions"
+  directory: "/"
+  schedule:
+    interval: "daily"
```
[the remark this answers is not in the document]
Comment (mattmoor) [reply .github/dependabot.yml]: done

File: .github/workflows/whitespace.yaml (hunk 2 above)
Comment (cpanato) [reply .github/workflows/whitespace.yaml:96]: 🙈

**Review by cpanato — APPROVED** [review]

**Pull request merged**